cmd/service_run.go

```diff
@@ -326,17 +339,48 @@ func startTokenDApp(log *zap.Logger, rf *RootFlags, openBrowser bool, cfg *netwo
 }
 
 // waitSig will wait for a sigterm or sigint interrupt.
-func waitSig(ctx context.Context, cfunc func(), log *zap.Logger) {
+func waitSig(ctx context.Context, cfunc func(), log *zap.Logger, pendingSigRequests chan service.ConsentRequest, p *printer.InteractivePrinter) {
 	gracefulStop := make(chan os.Signal, 1)
+
 	signal.Notify(gracefulStop, syscall.SIGTERM)
 	signal.Notify(gracefulStop, syscall.SIGINT)
 	signal.Notify(gracefulStop, syscall.SIGQUIT)
 
-	select {
-	case sig := <-gracefulStop:
-		log.Info("caught signal", zap.String("signal", fmt.Sprintf("%+v", sig)))
-		cfunc()
-	case <-ctx.Done():
-		// nothing to do
+	for {
+		select {
+		case sig := <-gracefulStop:
+			log.Info("caught signal", zap.String("signal", fmt.Sprintf("%+v", sig)))
+			cfunc()
+			return
+		case <-ctx.Done():
+			// nothing to do
+			return
+		case signRequest := <-pendingSigRequests:
+			txStr, err := signRequest.String()
+			if err != nil {
+				log.Info("failed to marshall sign request content")
+				cfunc()
+				return
+			}
+			p.CheckMark().Text("Received TX sign request: ").WarningText(txStr).NextLine()
+			reader := bufio.NewReader(os.Stdin)
+			p.CheckMark().WarningText("Please accept or decline sign request: (y/n)").NextLine()
+			answer, err := reader.ReadString('\n')
+			answer = strings.TrimSuffix(strings.TrimSuffix(answer, "\r\n"), "\n")
+			if err != nil {
+				log.Error("failed to read user input", zap.Error(err))
+				cfunc()
+				return
+			}
+			if answer == "y" || answer == "Y" {
+				log.Info("user approved sign request for transaction", zap.Any("transaction", txStr))
+				signRequest.Confirmations <- service.ConsentConfirmation{Decision: true, TxStr: txStr}
```

Comment:
You should close this single use channel now we're done with it.

Reply:
done